### DISCUSSION TITLE
Netbox configuration with Real IP instead of reverse-proxy IPs

### DISCUSSION THREAD

**Discussion by EkaterinaLatypova** [netbox-community/netbox#16362] [Help Wanted!]
Dear NetBox Support Team,

I hope this message finds you well.

I need help configuring NetBox to display real client IP addresses in the access logs instead of the IP address of our reverse proxy. Our reverse proxy is set up to forward the x-forwarded-host, x-forwarded-proto, and x-real-ip headers. However, NetBox does not seem to log these headers correctly.

Could you please provide guidance on the necessary configuration changes in NetBox to ensure it logs the real client IP addresses? Any insights or documentation would be greatly appreciated.

Netbox version 3.7.2

Thank you in advance for your support.

**Comment by candlerb** [comment]
> Our reverse proxy is set up to forward the x-forwarded-host, x-forwarded-proto, and x-real-ip headers

Are you sure? Run `tcpdump -i lo -nn -s0 -A tcp port 8001` on the Netbox server, then make a request. Then show *exactly* what headers it provides. (You may replace public IP addresses with dummy ones)

**Comment by markkuleinio** [comment]
I'm unsure which logging you are referring to. But here are my tips anyway:

These are for getting Nginx to log client source IPs in access.log: **/etc/nginx/conf.d/logging.conf**
```
# Access logging is globally disabled at http level...
access_log off;
# ... but it will be re-enabled at server levels in the enabled sites using this custom format:
log_format with_xff '$http_x_forwarded_for $remote_addr - $remote_user [$time_local] "$request" $status $body_bytes_sent "$http_referer" "$http_user_agent"';
```

and then add these in **/etc/nginx/sites-available/netbox** `server` section:

```
    # with_xff is defined in conf.d/logging.conf
    access_log /var/log/nginx/access.log with_xff;
```

This is also for **/etc/nginx/sites-available/netbox**, in the `location /` section, to get token IP restrictions to work with Gunicorn:
```
        # Works both with direct and load-balanced clients:
        proxy_set_header X-Real-IP $proxy_add_x_forwarded_for;
```
(If using uWSGI (and `uwsgi_pass` instead of `proxy_pass`), token IP restrictions work without specific header customization)

Works fine with AWS ELB doing source NAT in front of Nginx on NetBox server.
